soco/core.py

```diff
@@ -1833,6 +1873,185 @@ def add_item_to_sonos_playlist(self, queueable_item, sonos_playlist):
+                playlist, the length of resulting playlist, and the new
+                update_id.
+
+        Raises:
+            SoCoUPnPException: If object_id does not exist.
+        """
+        # allow either a string 'SQ:10' or an object with item_id attribute.
+        object_id = getattr(object_id, 'item_id', object_id)
+
+        if isinstance(tracks, UnicodeType):
+            track_list = [tracks, ]
+            position_list = [new_pos, ]
+        elif type(tracks) is int:
+            track_list = [tracks, ]
+            position_list = [new_pos or '', ]
+        else:
+            track_list = [str(x) for x in tracks]
+            position_list = [str(x) if x is not None else '' for x in new_pos]
+
+        if update_id == 0:  # retrieve the update id for the object
+            response, _ = self._music_lib_search(object_id, 0, 1)
+            update_id = response['UpdateID']
+        change = 0
+
+        for track, position in zip(track_list, position_list):
+            if track == position:   # there is no move, a no-op
+                continue
+            response = self.avTransport.ReorderTracksInSavedQueue([
+                ("InstanceID", 0),
+                ("ObjectID", object_id),
+                ("UpdateID", update_id),
+                ("TrackList", track),
+                ("NewPositionList", position),
+            ])
+            change += int(response['QueueLengthChange'])
+            update_id = int(response['NewUpdateID'])
+        length = int(response['NewQueueLength'])
+
+        return change, length, update_id
+
+    def clear_sonos_playlist(self, object_id, update_id=0):
+        """ Clear all tracks from a Sonos playlist.
+        This is a convenience function for :py:meth:`reorder_sonos_playlist`.
+
+        Example:
+            ``obj.clear_sonos_playlist('SQ:0')``
+
+        Args:
+            object_id (str): The item_id of the Sonos playlist or
+                the Sonos playlist object
+                (:py:class:`~.soco.data_structures.DidlPlaylistContainer`).
+            update_id (int): Optional update counter for the object. If left
+                at the default of 0, it will be looked up.
+
+        Returns:
+            tuple: See :py:meth:`reorder_sonos_playlist`
+
+        Raises:
+            SoCoUPnPException: See :py:meth:`reorder_sonos_playlist`
+        """
+        sonos_pl = None
+        if type(object_id) is DidlPlaylistContainer:
+            sonos_pl = object_id
+        else:
+            for spl in self.get_sonos_playlists():
+                if spl.item_id == object_id:
+                    sonos_pl = spl
+        if sonos_pl is None:
+            raise ValueError('"{0}" is not a valid Sonos Playlist'
+                             ' item_id'.format(object_id))
+        num_tracks = self.browse(ml_item=sonos_pl).total_matches
+        tracks = ','.join([str(x) for x in range(num_tracks)])
+        return self.reorder_sonos_playlist(sonos_pl, tracks=tracks,
```

Comment:
O.k. I see here that you use the non-iterable input form, but it would only be one extra line not to, if it were to be removed. If this works however, this does seem to indicate that one request is handled as one *query* on the other end, otherwise, this would not work, it would run into non-existing positions along th way.